### DISCUSSION TITLE
Add a new modifier to check if parent component is loading

### DISCUSSION THREAD

**Discussion by slovenianGooner** [livewire/livewire#9093] [3. Feature ideas]
Let me present a simple scenario :)

I have a Table component which renders the table header, some rows and has pagination.

Each row is it's own Livewire component called TableRow.

What I want to achieve is to use wire:loading on a TableRow component to display a "skeleton" when the parent table is loading (pagination, filtering, sorting, etc.). Of course, with the current wire:loading directive that is not possible, it only detects loading from the current component.

Maybe this is already possible and I'm missing something from the docs, but I did a big research, even source diving into JS of Livewire and couldn't find a solution.

So, the suggestion is: add a new modifier for wire:loading, that would signal to the component to listen for parent loading state.

`wire:loading.parent`

Is something like that even possible?

**Comment by lrljoe** [comment]
Why on **earth** would you have each row of a table as a separate livewire component?

That aside, "wire:target" exists to target an action or property update.

I think you're going about it a bit strangely to be honest, and given that each Livewire Component is an island, albeit with Modelable capabilities to provide interactivity between parent/child.  This kind of behaviour would introduce such an overhead that 99.9% of developers wouldn't ever use.

**Reply by slovenianGooner** [reply]
To be fair, it's a concept even Caleb uses in one of his screencasts for Flux tables. 

I use it because each row includes a form with 3 fields (3 columns) and the handling of that is done in the table row component itself. 

And yes, I am aware of the wire:target directive which I heavily use as well. 

Problem is I want to display a skeleton on the table row component when I change pages or when I filter results on the table component - and since like you said, each Livewire component is an island, the table row is not aware of table being wire:loading. 

To be fair, it's an obscure case even in my app, because I mostly have HTML directly in the table component, but there are times when there is a need for a separate component. 👍

Please see the solution I came up with, it's very simple: 
https://github.com/livewire/livewire/pull/9094

**Comment by swen4** [comment]
@slovenianGooner Did you end up finding a solution for this? Running into the exact same issue! I have a table where each row is a livewire component. With livewire pagination, I want to display scaffold rows until the new page is displayed.
